### ISSUE TITLE
Exclusions must be given relative to the config file, not the package -- intended?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2780]
```r
options(lintr.linter_file = tmp<-tempfile())
with_tempdir({
  cat(sprintf("Linting from %s\n", getwd()))

  writeLines('a=1', 'a.R')
  writeLines('a=1', 'b.R')

  cat("First pass:\n")
  writeLines('exclusions: list("a.R")', tmp)
  print(lint_dir())

  cat("Second pass:\n")
  A.R <- normalizePath("a.R")
  writeLines(paste0('exclusions: list("', A.R, '")'), tmp)
  print(lint_dir())
})
```

Has output:

```
Linting from /tmp/RtmpRm034g/fileac1f1d3f2a83
First pass:
a.R:1:2: style: [assignment_linter] Use one of <-, <<- for assignment, not =.
a=1
 ^
a.R:1:2: style: [infix_spaces_linter] Put spaces around all infix operators.
a=1
 ^
b.R:1:2: style: [assignment_linter] Use one of <-, <<- for assignment, not =.
a=1
 ^
b.R:1:2: style: [infix_spaces_linter] Put spaces around all infix operators.
a=1
 ^
Second pass:
b.R:1:2: style: [assignment_linter] Use one of <-, <<- for assignment, not =.
a=1
 ^
b.R:1:2: style: [infix_spaces_linter] Put spaces around all infix operators.
a=1
 ^
```

That is, the path relative to the directory where `lint_dir()` is run, 'a.R', is not picked up as being excluded in the first path, hence we see lints on a.R there.

By contrast, when we give the _full_ absolute path to a.R in the second pass, it's correctly excluded.

Is this intended behavior? If so I think it needs to be more clearly documented.

It would certainly be clearer in my use case to keep using paths relative to the directory where `lint()` is to be run:

 - {data.table} keeps .ci/.lintr.R out of the top-level directory to keep that a bit tidier
 - but paths in the config must use `../` to apply correctly: https://github.com/Rdatatable/data.table/blob/f78044ebc63cc70a7ef67607b66fc8158766be86/.ci/.lintr.R#L114

**Comment by AshesITR** [comment]
See #580

https://github.com/r-lib/lintr/issues/580#issuecomment-737984557

The behavior is intentional (so that `expect_lint_free()` works when run from within the `tests/` subdirectory of a package.

`.lintr.R` has access to `getwd()` if desired.
